bioptim/dynamics/dynamics_functions.py

```diff
@@ -63,10 +63,12 @@ def custom(states: MX.sym, controls: MX.sym, parameters: MX.sym, nlp) -> MX:
         ----------
         MX.sym
             The derivative of the states
+        MX.sym
+            The defects of the implicit dynamics
         """
 
-        qdot, qddot = nlp.dynamics_type.dynamic_function(states, controls, parameters, nlp)
-        return vertcat(qdot, qddot)
+        xdot, defects = nlp.dynamics_type.dynamic_function(states, controls, parameters, nlp)
+        return xdot, defects
```

Comment:
This should also take into account the dynamics when the user does not provide the implicit version.